### PR TITLE
Added vmware route priority

### PR DESCRIPTION
**Pull request by BShrivastav**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #[UI-24161](https://jiracloud.swg.usma.ibm.com:8443/browse/UI-24161)

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: ibm/service/vpc/resource_ibm_is_vpc_routing_table_route.go [hunk 1]
```diff
@@ -66,9 +66,9 @@ func ResourceIBMISVPCRoutingTableRoute() *schema.Resource {
 				Description: "The zone to apply the route to. Traffic from subnets in this zone will be subject to this route.",
 			},
 			rNextHop: {
-				Type:        schema.TypeString,
-				Required:    true,
-				ForceNew:    true,
+				Type:     schema.TypeString,
+				Required: true, // recheck as the generated says it as, not required.
```
Comment (deepaksibm) [inline ibm/service/vpc/resource_ibm_is_vpc_routing_table_route.go:70]: I see it is not required field in the spec, but if it is required already we can leave it as it is.

File: website/docs/r/is_vpc_routing_table_route.html.markdown [hunk 2]
```diff
@@ -82,6 +96,7 @@ In addition to all argument reference list, you can access the following attribu
 - `lifecycle_state` - (String) The lifecycle state of the route.
 - `origin` - (Optional, String) The origin of this route:- `service`: route was directly created by a service- `user`: route was directly created by a userThe enumerated values for this property are expected to expand in the future. When processing this property, check for and log unknown values. Optionally halt processing and surface the error, or bypass the route on which the unexpected property value was encountered.
   - Constraints: Allowable values are: `learned`, `service`, `user`.
+- `priority` - (Integer) The route's priority. Smaller values have higher priority. If a routing table contains routes with the same destination, the route with the highest priority (smallest value) is selected. For Example (2), supports values from 0 to 255.
```
Comment (deepaksibm) [inline website/docs/r/is_vpc_routing_table_route.html.markdown:99]: add 'optional'

Comment (BShrivastav) [reply website/docs/r/is_vpc_routing_table_route.html.markdown]: modified.

File: ibm/service/vpc/resource_ibm_is_vpc_routing_table_route.go [hunk 3]
```diff
@@ -156,6 +155,12 @@ func ResourceIBMISVPCRoutingTableRoute() *schema.Resource {
 				Computed:    true,
 				Description: "Routing table route Lifecycle State",
 			},
+			"priority": {
```
[the remark this answers is not in the document]
Comment (BShrivastav) [reply ibm/service/vpc/resource_ibm_is_vpc_routing_table_route.go:158]: this is added to support the existing range.

Screenshot 2022-09-21 at 11 09 59 AM

File: ibm/service/vpc/resource_ibm_is_vpc_routing_table_route.go [hunk 4]
```diff
@@ -265,7 +284,6 @@ func resourceIBMISVPCRoutingTableRouteRead(d *schema.ResourceData, meta interfac
 
 	d.Set(rID, *route.ID)
 	d.Set(rName, *route.Name)
-	d.Set(rDestination, *route.Destination)
```
Comment (deepaksibm) [inline ibm/service/vpc/resource_ibm_is_vpc_routing_table_route.go:268]: are we removing this ?

Comment (BShrivastav) [reply ibm/service/vpc/resource_ibm_is_vpc_routing_table_route.go:268]: reverted.

**Comment by BShrivastav** [comment]
merged: https://github.com/IBM-Cloud/terraform-provider-ibm/pull/4435

**Pull request closed without merging**